packages/integration-tests/src/logto-context.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  generateCodeChallenge,
+  generateCodeVerifier,
+  generateSignInUri,
+  generateState,
+} from '@logto/js';
+import got, { Response } from 'got/dist/source';
+
+import { authorizationEndpoint, adminConsoleAppId, redirectUri } from './constants';
+import { extractCookie } from './utils';
+
+type Account = {
+  username: string;
+  password: string;
+};
+
+export class LogtoContext {
+  private codeVerifier = '';
+  private codeChallenge = '';
+  private state = '';
+  private cookie = '';
+
+  constructor(private readonly account: Account) {}
+
+  public async startInteraction() {
+    this.codeVerifier = generateCodeVerifier();
+    this.state = generateState();
+    this.codeChallenge = await generateCodeChallenge(this.codeVerifier);
+
+    const signInUri = generateSignInUri({
+      authorizationEndpoint,
+      clientId: adminConsoleAppId,
+      redirectUri,
+      codeChallenge: this.codeChallenge,
+      state: this.state,
+    });
+
+    const response: Response = await got(signInUri, {
+      followRedirect: false,
+    });
+
+    this.cookie = extractCookie(response);
+  }
+
+  public getCookie() {
+    return this.cookie;
+  }
+
+  public updateCookie(cookie: string) {
+    this.cookie = cookie;
+  }
+
+  public getCodeVerifier() {
+    return this.codeVerifier;
+  }
+
+  public getState() {
+    return this.state;
+  }
```

Comment:
maybe make them public or use js property getter?
seems that there is no need to encapsulate these simple getters like java/kotlin?

Reply:
updated